lib/std/fs/File.zig

```diff
@@ -1195,20 +1195,20 @@ pub fn readAll(self: File, buffer: []u8) ReadError!usize {
     return index;
 }
 
-/// On Windows, this function currently does alter the file pointer.
-/// https://github.com/ziglang/zig/issues/12783
 pub fn pread(self: File, buffer: []u8, offset: u64) PReadError!usize {
     if (is_windows) {
-        return windows.ReadFile(self.handle, buffer, offset);
+        // On Windows, this would alter the file pointer, so reset it.
+        const curr_pos = self.getPos() catch return error.Unseekable;
+        const n = try windows.ReadFile(self.handle, buffer, offset);
```

Comment:
I'm unsure if an error in ReadFile will leave the seek position changed, perhaps it should be reset in the case of an error, or reset in a defer.

Reply:
My reading of the docs is that the file pointer is only updated on success:

https://learn.microsoft.com/en-us/windows/win32/api/fileapi/nf-fileapi-readfile#synchronization-and-file-position

(see *Considerations for working with synchronous file handles:*)